### PR TITLE
Check user command

### PR DESCRIPTION
**Pull request by RishiVaya**
added validation for users and their command checks

added helper methods for admin checks

### CODE REVIEW

File: src/leo-server-app/src/routes/schedule.ts [hunk 1]
```diff
@@ -103,21 +104,40 @@ async function validateCommands(satelliteId: string, commands: string[]) {
...
 
+const isAdminCheck = async (userId: string) => {
+  const userRecord = await User.findById(userId);
+  console.log(userRecord?.role);
+  return userRecord?.role === UserRole.ADMIN;
+};
+
+async function verifyUserCommands(satelliteId: string, userId: string, commands: string[]) {
+  // Get satellite data
+  // const satellite = await Satellite.findById(satelliteId).exec();
```
Comment (quinnha) [inline src/leo-server-app/src/routes/schedule.ts:115]: remove comment if not needed

File: src/leo-server-app/src/routes/schedule.ts [hunk 2]
```diff
@@ -103,21 +104,40 @@ async function validateCommands(satelliteId: string, commands: string[]) {
...
     body.satelliteId,
+    body.userId,
     body.commands
   );
 
   let resObj = {};
 
-  if (!isCommandsValid) {
+  const adm = await isAdminCheck(body.userId);
+    console.log(isCommandsValid, adm)
```
Comment (quinnha) [inline src/leo-server-app/src/routes/schedule.ts:137]: is this needed?

File: src/leo-server-app/src/routes/schedule.ts [hunk 3]
```diff
@@ -139,22 +159,16 @@ router.post("/createSchedule", async (req: CreateScheduleProp, res: any) => {
...
+// ) => {
+//   // Fetch satellite
+//   const satellite = await Satellite.findById(satelliteId);
 
-  const isValid = satellite?.validCommands.includes(command);
-  return isValid;
-};
+//   const isValid = satellite?.validCommands.includes(command);
+//   return isValid;
+// };
```
Comment (quinnha) [inline src/leo-server-app/src/routes/schedule.ts:171]: remove if commented

**Review by quinnha — COMMENTED** [review]
just make sure tests are passing then i'll give the lgtm

File: src/leo-server-app/src/routes/schedule.ts [hunk 4]
```diff
@@ -103,21 +104,40 @@ async function validateCommands(satelliteId: string, commands: string[]) {
...
+
+async function verifyUserCommands(satelliteId: string, userId: string, commands: string[]) {
+  // Get satellite data
+  // const satellite = await Satellite.findById(satelliteId).exec();
+  const comms = await validateUserCommands(satelliteId, userId)
+  console.log(comms)
+
+  // Check if commands are valid
+  return commands.every((cmd) => comms?.includes(cmd));
+}
```
Comment (Umang-Rajkarnikar) [inline src/leo-server-app/src/routes/schedule.ts:121]: Can you move these to the util file

**Review by quinnha — APPROVED** [review]
lgtm

File: src/leo-server-app/package.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
     "build": "npx tsc",
     "start:dev": "cross-env NODE_ENV=development nodemon src/server.ts",
     "all": "concurrently \"cd ../leo-client-app && npm run dev\" \"npm run start:dev\"",
-    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js --silent"
+    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js"
```
Comment (Umang-Rajkarnikar) [inline src/leo-server-app/package.json:10]: revert back to silent

**Review by Umang-Rajkarnikar — APPROVED** [review]
lgtm, just address comment

**Pull request merged**